lib/cli/src/frameworks/react/ts/Welcome.tsx

```diff
@@ -0,0 +1,159 @@
+import React, {
+  AnchorHTMLAttributes,
+  ButtonHTMLAttributes,
+  DetailedHTMLProps,
+  FunctionComponent,
+  HTMLAttributes,
+} from 'react';
+
+type MainProps = Omit<DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>, 'style'>;
+const Main: FunctionComponent<MainProps> = (props) => (
+  <article
+    {...props}
+    style={{
+      padding: 15,
+      lineHeight: 1.4,
+      fontFamily: '"Helvetica Neue", Helvetica, "Segoe UI", Arial, freesans, sans-serif',
+      backgroundColor: '#fff',
+      color: '#000',
+    }}
+  />
+);
+
+type TitleProps = DetailedHTMLProps<HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>;
+const Title: FunctionComponent<TitleProps> = ({ children, ...props }) => (
+  <h1 {...props}>{children}</h1>
+);
+
+type NoteProps = Omit<
+  DetailedHTMLProps<HTMLAttributes<HTMLParagraphElement>, HTMLParagraphElement>,
+  'style'
+>;
+const Note: FunctionComponent<NoteProps> = (props) => (
+  <p
+    {...props}
+    style={{
+      opacity: 0.5,
+    }}
+  />
+);
+
+type InlineCodeProps = Omit<DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>, 'style'>;
+const InlineCode: FunctionComponent<InlineCodeProps> = (props) => (
+  <code
+    {...props}
+    style={{
+      fontSize: 15,
+      fontWeight: 600,
+      padding: '2px 5px',
+      border: '1px solid #eae9e9',
+      borderRadius: 4,
+      backgroundColor: '#f3f2f2',
+      color: '#3a3a3a',
+    }}
+  />
+);
+
+type LinkProps = Omit<
+  DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>,
+  'style'
+> & {
+  href: string;
+  target: string;
+  rel: string;
+};
+const Link: FunctionComponent<LinkProps> = ({ children, href, target, rel, ...props }) => (
+  <a
+    href={href}
+    {...props}
+    target={target}
+    rel={rel}
+    style={{
+      color: '#1474f3',
+      textDecoration: 'none',
+      borderBottom: '1px solid #1474f3',
+      paddingBottom: 2,
+    }}
+  >
+    {children}
+  </a>
+);
+
+type NavButtonProps = Omit<
+  DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
+  'style' | 'type'
+> & {};
+const NavButton: FunctionComponent<NavButtonProps> = ({ children, onClick, ...props }) => (
+  <button
+    {...props}
+    type="button"
+    onClick={onClick}
+    style={{
+      color: '#1474f3',
+      textDecoration: 'none',
+      borderBottom: '1px solid #1474f3',
+      paddingBottom: 2,
+      borderTop: 'none',
+      borderRight: 'none',
+      borderLeft: 'none',
+      backgroundColor: 'transparent',
+      padding: 0,
+      cursor: 'pointer',
+      font: 'inherit',
+    }}
+  >
+    {children}
+  </button>
+);
+
+interface WelcomeProps {
+  showApp: () => void;
+}
+const Welcome: FunctionComponent<WelcomeProps> = ({ showApp }) => (
```

Comment:
Can we add `{ showApp }: WelcomeProps` here? It's currently required if the user switches to `react-docgen` for props tables. It's not the default, but I expect lots of users will try switching and immediately the props will disappear without that type annotation due to https://github.com/reactjs/react-docgen/issues/387